USER_GUIDE.md

```diff
@@ -123,20 +123,24 @@ Test Scenario uses Test description from the previous step. Below is the `myconf
 ```toml
 name = "nccl-test"
 
-[Tests.1]
-  name = "nccl_test_all_reduce_single_node"
-  time_limit = "00:20:00"
-
-[Tests.2]
-  name = "nccl_test_all_reduce_single_node"
-  time_limit = "00:20:00"
-  [Tests.2.dependencies]
-    start_post_comp = { name = "Tests.1", time = 0 }
+[[Tests]]
+id = "Tests.1"
+template_test = "nccl_test_all_reduce_single_node"
+time_limit = "00:20:00"
```

Comment:
Let's discuss the new field names in the design meeting (id, template_test). 'ID' looks good to me, but I prefer 'test_definition' over 'template_test'.

Reply:
Make sense, let's discuss.

Reply:
I gave `test_definition` a though and I think that it is misleading in this context: we reference a particular Test TOML which is based on a `TestDefinition`, but we do not reference definition directly.

Reply:
This PR looks good to me, except for the field naming. If others are okay with the field name, I can approve.

Reply:
Why not test_name? We use 'name' in the test toml.

Reply:
`test_name` sounds good to me.
@TaekyungHeo @srivatsankrishnan what do you think?

Reply:
`tests_name` looks good.

Reply:
Went ahead and renamed it here and in Cloud AI X as well.

Reply:
test_name looks good to me as well.